tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
+
+    my_evaluation_dependency_1_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    dependencies_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    my_dependent_parameter_builder.resolve_evaluation_dependencies(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        fully_qualified_parameter_names=dependencies_fully_qualified_parameter_names,
+        runtime_configuration=None,
+    )
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    These assertions show that both "ParameterBuilder" dependencies computed their values, but dependent
+    "ParameterBuilder" has not yet computed its value.
+    """
+    assert (
+        my_evaluation_dependency_0_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_0_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+    assert (
+        my_evaluation_dependency_1_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_1_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+    assert (
+        my_dependent_parameter_builder.raw_fully_qualified_parameter_name
+        not in dependencies_fully_qualified_parameter_names
+        or my_dependent_parameter_builder.json_serialized_fully_qualified_parameter_name
+        not in dependencies_fully_qualified_parameter_names
+    )
+    """
+    No "ParameterBuilder" dependencies need to be executed, because None are specified (hence,
+    "my_dependent_parameter_builder.resolve_evaluation_dependencies()" has no effect), but their values exist, because
+    they were executed individually previously.  Thus, set of computed fully_qualified_parameter_names does not change.
+    """
+    assert sorted(all_fully_qualified_parameter_names) == sorted(
+        dependencies_fully_qualified_parameter_names
+    )
+
+    my_dependent_parameter_builder.build_parameters(
```

Comment:
I understand this to be the method under test: assertions should immediately follow its invocation to ensure the proper behavior. What can we assert after calling `build_parameters` that doesn't require me to invoke other methods?

Reply:
@donaldheppner Got it -- I moved relevant invocation closer to assertions so that it is easier to read.  Thank you.